clarity_ext/utility/testing_parse_scripts/builders.py

```diff
@@ -0,0 +1,122 @@
+from mock import MagicMock
+from StringIO import StringIO
+from io import BytesIO
+from clarity_ext.service.file_service import Csv
+from clarity_ext.service.file_service import FileService
+from clarity_ext.service.process_service import ProcessService
+from clarity_ext.context import ExtensionContext
+from clarity_ext.service.artifact_service import ArtifactService
+from clarity_ext.domain.process import Process
+from clarity_ext.domain.user import User
+from clarity_ext.domain.udf import UdfMapping
+from clarity_ext.utility.testing_parse_scripts.fake_artifact_factory import FakeArtifactFactory
+
+
+class ContextBuilder:
+    """
+    Add entities to repositories held by context
+    E.g. shared files, artifacts, analytes.
+    """
+    def __init__(self):
+        self.step_repo = FakeStepRepo()
+        self.logger = FakeLogger()
+        self.context = None
+        self._create()
+
+    def with_analyte_pair(self, input, output):
+        self.step_repo.add_analyte_pair(input, output)
+
+    def with_mocked_local_shared_file(self, filename, contents):
+        monkey = LocalSharedFilePatcher()
+        monkey.cache[filename] = contents
+        self.context.local_shared_file = monkey.local_shared_file
+
+    def _create(self):
+        session = MagicMock()
+        file_repository = MagicMock()
+        artifact_service = ArtifactService(self.step_repo)
+        os_service = MagicMock()
+        current_user = None
+        file_service = FileService(
+            artifact_service, file_repository, False,
+            os_service, uploaded_to_stdout=False, disable_commits=True)
+        validation_service = MagicMock()
+        dilution_service = MagicMock()
+        process_service = ProcessService()
+        clarity_service = None
+        self.context = ExtensionContext(session, artifact_service, file_service, current_user,
+                                self.logger, self.step_repo, clarity_service,
+                                dilution_service, process_service, validation_service,
+                                test_mode=False, disable_commits=True)
+
+
+class PairBuilder(object):
+    def __init__(self):
+        self.artifact_repo = FakeArtifactFactory()
+        self.output_udf_dict = dict()
+        self.target_id = None
+        self.target_type = None
+        self.pair = None
+
+    def create(self):
+        pair = self.artifact_repo.create_pair(
+            pos_from=None, pos_to=None, source_id=None, target_id=self.target_id,
+            target_type=self.target_type)
+        pair.output_artifact.udf_map = UdfMapping(self.output_udf_dict)
+        self.pair = pair
+
+    def with_target_id(self, target_id):
```

Comment:
I not sure what convention has been using in clarity-ext, but normally I'd expect these types of methods to be called `set_*`. I don't object to this, I just wanted to check that there is a reasoning behind it.

Reply:
:+1:

Reply:
That makes sense!